semgrep/semgrep/metrics.py

```diff
@@ -287,7 +287,6 @@ def send(self) -> None:
                 # TODO: weird, we get a mypy error only in CI, and only
                 # after I've enabled submodules for pre-commit
                 headers=cast(Dict[str, str], {"Authorization": None}),
```

Comment:
@nbrahms we already have a default timeout of 30s that comes from the app session. I think the timeout of 2s might be excessively low and might cause missed metrics (considering that most of my request timings were around 2.5s, and that's from our gigabit office internet right next to the AWS datacenter.)

I'd be ok with a middle ground timeout, but since at this point the results have been printed already, I thought it'd be fine to wait as long as for any other request, and thus end up with fewer special cases.

Reply:
We should find the original history; the intention was to prefer dropping metrics over blocking the command line for more than 2 seconds. (cc. @ievans ; I'm not sure what the ideal number is, but intuitively it should be far lower than 30 s)

Reply:
what if we launch a new `semgrep` subprocess to send off the metrics so that the scan process can exit immediately?

Reply:
yup, just need that daemonized